modules/core/src/main/java/org/apache/ignite/internal/hlc/HybridTimestamp.java

```diff
@@ -190,6 +220,6 @@ public String toString() {
      * Returns a new hybrid timestamp with incremented physical component.
      */
     public HybridTimestamp addPhysicalTime(long mills) {
-        return new HybridTimestamp(physical + mills, logical);
+        return new HybridTimestamp(addExact(getPhysical(), mills), getLogical());
```

Comment:
Wouldn't it be more effective to shift `mills` and add it to the whole long value?